### DISCUSSION TITLE
question: custom layout does what exactly?

### DISCUSSION THREAD

**Discussion by proddy** [table-library/react-table-library#53] [Q&A] [answered]
in <Table> what does the attribute `layout` do when it's set to true `<Table .... layout={{ custom: true }} ... `

the reason I'm asking is that I set `layout custom:true` for some fancy table formatting I need but found when the browser window resizes it doesn't re-calculate and re-align the columns correctly unless forced by a F5 for example.

**Comment by rwieruch** [comment, marked as answer]
Only when `layout={{ custom: true }}` is set, one can give the cells in the theme a custom width:

```
    const theme = useTheme({
      BaseCell: `
        &:nth-of-type(1) {
          min-width: 35%;
          width: 35%;
        }

        &:nth-of-type(2), &:nth-of-type(3), &:nth-of-type(4) {
          min-width: 15%;
          width: 15%;
        }

        &:nth-of-type(5) {
          min-width: 20%;
          width: 20%;
        }
      `,
    });
```

If it is not set, the cells are just divided equally.

**Reply by proddy** [reply]
thanks! I get it now. I'm having trouble with auto-sizing when the window is resized but will open a new issue for that with some sample code

**Reply by proddy** [reply]
https://github.com/table-library/react-table-library/issues/55
